std/format.d

```diff
@@ -2669,7 +2676,7 @@ if (is(FloatingPointTypeOf!T) && !is(T == enum) && !hasToString!(T, Char))
         "floating point formatting failure");
 
     size_t len = min(n, buf.length-1);
-    if (fs.flSeparator)
+    if (fs.flSeparator && !isNaN(val))
```

Comment:
Why this check? I think nan is allready ruled out above, isn't it?

Reply:
true, I'll fix that

Reply:
fixed